File: csharp/winforms/PluginSample/PluginSample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace PluginSample
{

  public partial class Form1 : Form
  {

    private string _CurrentFile = "";

    public Form1()
    {
      InitializeComponent();
    }

    private void mnuNew_Click(object sender, EventArgs e)
    {
      textBox1.Text = "";
    }

    private void mnuOpen_Click(object sender, EventArgs e)
    {
      if (openFileDialog1.ShowDialog() == DialogResult.OK)
      {
        _CurrentFile = openFileDialog1.FileName;
        System.IO.StreamReader myRead = new System.IO.StreamReader(_CurrentFile, System.Text.Encoding.GetEncoding(1251));
        textBox1.Text = myRead.ReadToEnd();
        myRead.Close();
      }
    }

    private void mnuSave_Click(object sender, EventArgs e)
    {
      if (saveFileDialog1.ShowDialog() == DialogResult.OK)
      {
        _CurrentFile = saveFileDialog1.FileName;
        System.IO.StreamWriter myWrite = new System.IO.StreamWriter(_CurrentFile, false, System.Text.Encoding.GetEncoding(1251));
        myWrite.Write(textBox1.Text);
        myWrite.Close();
      }
    }

    private void mnuExit_Click(object sender, EventArgs e)
    {
      this.Close();
    }

    private void Form1_FormClosing(object sender, System.Windows.Forms.FormClosingEventArgs e)
    {
      if (e.CloseReason == CloseReason.UserClosing)
      {
        if (MessageBox.Show("Are you want to exit?", "Exit", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.No)
        // if (MessageBox.Show("Вы действительно хотие выйти из программы?", "Выход?", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.No)
        {
          e.Cancel = true;
        }
      }
    }

    private void Form1_Load(object sender, EventArgs e)
    {
      openFileDialog1.Filter = "Text files (*.txt)|*.txt|All files (*.*)|*.*";
      saveFileDialog1.Filter = "Text files (*.txt)|*.txt|All files (*.*)|*.*";
      // openFileDialog1.Filter = "Текстовые документы (*.txt)|*.txt|Все файлы (*.*)|*.*";
      // saveFileDialog1.Filter = "Текстовые документы (*.txt)|*.txt|Все файлы (*.*)|*.*";
      LoadPlugins();
    }

    private void LoadPlugins()
    {
      string sPath = Application.StartupPath;
      List<IPlugin> lPlugins = new List<IPlugin>();

      foreach (string f in System.IO.Directory.GetFiles(sPath, "*.dll"))
      {
        System.Reflection.Assembly a = System.Reflection.Assembly.LoadFile(f);

        foreach (Type t in a.GetTypes())
        {
          foreach (Type i in t.GetInterfaces())
          {
            if (i.Equals(Type.GetType("PluginSample.IPlugin")))
            {
              IPlugin p = (IPlugin)Activator.CreateInstance(t);
              lPlugins.Add(p);
              p.Initialize(this);
              break;
            }
          }
        }
      }
    }

  }

}

File: examples/vbnet/winforms/PluginSample/PluginSample/Form1.vb
Public Class Form1

  Private _CurrentFile As String = ""

  Private Sub mnuNew_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mnuNew.Click
    TextBox1.Text = ""
  End Sub

  Private Sub mnuOpen_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mnuOpen.Click
    If OpenFileDialog1.ShowDialog = Windows.Forms.DialogResult.OK Then
      _CurrentFile = OpenFileDialog1.FileName
      Dim myRead As New IO.StreamReader(_CurrentFile, System.Text.Encoding.GetEncoding(1251))
      TextBox1.Text = myRead.ReadToEnd
      myRead.Close()
    End If
  End Sub

  Private Sub mnuSave_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mnuSave.Click
    If SaveFileDialog1.ShowDialog = Windows.Forms.DialogResult.OK Then
      _CurrentFile = SaveFileDialog1.FileName
      Dim myWrite As New IO.StreamWriter(_CurrentFile, False, System.Text.Encoding.GetEncoding(1251))
      myWrite.Write(TextBox1.Text)
      myWrite.Close()
    End If
  End Sub

  Private Sub mnuExit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mnuExit.Click
    Me.Close()
  End Sub

  Private Sub Form1_FormClosing(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles Me.FormClosing
    If e.CloseReason = CloseReason.UserClosing Then
      If MsgBox("Are you want to exit?", MsgBoxStyle.Question Or MsgBoxStyle.YesNo) = MsgBoxResult.No Then e.Cancel = True
      'If MsgBox("Вы действительно хотие выйти из программы?", MsgBoxStyle.Question Or MsgBoxStyle.YesNo) = MsgBoxResult.No Then e.Cancel = True
    End If
    End
  End Sub

  Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
    OpenFileDialog1.Filter = "Text files (*.txt)|*.txt|All files (*.*)|*.*"
    SaveFileDialog1.Filter = "Text files (*.txt)|*.txt|All files (*.*)|*.*"
    'OpenFileDialog1.Filter = "Текстовые документы (*.txt)|*.txt|Все файлы (*.*)|*.*"
    'SaveFileDialog1.Filter = "Текстовые документы (*.txt)|*.txt|Все файлы (*.*)|*.*"
    LoadPlugins()
  End Sub

  'процедура подключения плагинов
  Private Sub LoadPlugins()
    Dim sPath As String = Application.StartupPath
    Dim lPlugins As New List(Of IPlugin)
    For Each f As String In IO.Directory.GetFiles(sPath, "*.dll")
      Dim a As Reflection.Assembly = Reflection.Assembly.LoadFile(f)
      For Each t As Type In a.GetTypes()
        For Each i As Type In t.GetInterfaces()
          If i.Equals(GetType(IPlugin)) Then
            Dim p As IPlugin = CType(Activator.CreateInstance(t), IPlugin)
            lPlugins.Add(p)
            p.Initialize(Me)
            Exit For
          End If
        Next
      Next
    Next
  End Sub

End Class